core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -752,9 +760,15 @@ public TableMetadata buildReplacement(Schema updatedSchema, PartitionSpec update
       sortOrdersBuilder.add(freshSortOrder);
     }
 
-    Map<String, String> newProperties = Maps.newHashMap();
-    newProperties.putAll(this.properties);
-    newProperties.putAll(updatedProperties);
+    // prepare new table properties
+    Map<String, String> newRawProperties = Maps.newHashMap();
+    newRawProperties.putAll(this.properties);
+    newRawProperties.putAll(updatedProperties);
+    Map<String, String> newProperties = unreservedProperties(newRawProperties);
```

Comment:
I think it would be better to construct the properties directly rather than building a map and then rebuilding it. You should be able to use the same `filter` to add just the non-reserved properties from `updatedProperties`.

Reply:
You could have `unreservedProperties` return something else, like a filtered `Stream` if you wanted to still reuse it.